CMakeLists.txt

```diff
@@ -143,6 +143,9 @@ else()
   endforeach(flag_var)
 endif()
 
+set (CMAKE_CXX_FLAGS_DEBUG "${CMAKE_CXX_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
+set (CMAKE_LINKER_FLAGS_DEBUG "${CMAKE_STATIC_LINKER_FLAGS_DEBUG} -fno-omit-frame-pointer -fsanitize=address")
```

Comment:
Do we want to enable it for all cases?

Reply:
if by all cases you mean all debug builds, I'd argue yes :)

Reply:
Yeah I was mainly asking for whether there should be a flag (although this flag should be ON by default) to disable ASAN because I have seen cases that people need to temporarily disable ASAN build to unblock (this can happen when ASAN has false positive, or there are some bad changes have sneaked into master).